src/app/view/components/dot-add-persona-dialog/dot-create-persona-form/dot-create-persona-form.component.ts

```diff
@@ -0,0 +1,117 @@
+    @Output() isValid: EventEmitter<Boolean> = new EventEmitter();
+
+    form: FormGroup;
+    imageName: string;
+    messagesKey: { [key: string]: string } = {};
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        public dotMessageService: DotMessageService,
+        private fb: FormBuilder,
+        private siteService: SiteService
+    ) {}
+
+    ngOnInit() {
+        this.dotMessageService
+            .getMessages([
+                'modes.persona.upload.file',
+                'modes.persona.name',
+                'modes.persona.key.tag',
+                'dot.common.choose',
+                'dot.common.remove',
+                'modes.persona.host',
+                'modes.persona.name.error.required'
+            ])
+            .pipe(take(1))
+            .subscribe((messages: { [key: string]: string }) => {
+                this.messagesKey = messages;
+            });
+        this.initPersonaForm();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    /**
+     * Handle the response of the p-fileUpload to update the form.
+     *
+     * @param {DotFileUpload} event
+     * @memberof DotCreatePersonaFormComponent
+     */
+    onFileUpload(event: DotFileUpload) {
+        const response = JSON.parse(event.xhr.response);
+        const tempFile: DotCMSTempFile = response.tempFiles[0];
+        this.imageName = event.files[0].name;
+        this.form.get('photo').setValue(tempFile.id);
+    }
+
+    /**
+     * Remove selected image.
+     *
+     * @memberof DotCreatePersonaFormComponent
+     */
+    removeImage(): void {
+        this.imageName = null;
+        this.form.get('photo').setValue('');
+    }
+
+    /**
+     * Update the site identifier in the form.
+     *
+     * @param {Site} site
+     * @memberof DotCreatePersonaFormComponent
+     */
+    siteChange(site: Site) {
+        this.form.get('hostFolder').setValue(site.identifier);
+    }
+
+    /**
+     * Set the key tag attribute with camelCase standard based on the name.
+     *
+     * @memberof DotCreatePersonaFormComponent
+     */
+    setKeyTag(): void {
+        this.form.get('keyTag').setValue(_.camelCase(this.form.get('name').value));
+    }
+
+    /**
+     * Reset form to default values
+     *
+     * @memberof DotCreatePersonaFormComponent
+     */
+    resetForm(): void {
+        this.imageName = null;
+        this.form = this.fb.group({
+            hostFolder: [this.siteService.currentSite.identifier, [Validators.required]],
+            keyTag: [{ value: '', disabled: true }, [Validators.required]],
+            name: ['', [Validators.required]],
+            photo: ['', [Validators.required]]
+        });
+    }
+
+    private initPersonaForm(): void {
+        this.resetForm();
+        this.form.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(() => {
+            this.isValid.emit(this.form.valid);
```

Comment:
If you already have access to the whole form from the outside why you need to emit the value of valid if you can just get it directly from the form outside.

Reply:
will use this to emit false while waiting  the temp file upload response. So now make sense to keep it.